Explorer/Assets/DCL/Multiplayer/SDK/Systems/WritePlayerIdentityDataSystem.cs

```diff
@@ -0,0 +1,87 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CRDT;
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Multiplayer.SDK.Components;
+using ECS.Abstract;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Unity.Groups;
+using UnityEngine;
+using Query = Arch.Core.Query;
+
+namespace DCL.Multiplayer.SDK.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.PLAYER_IDENTITY_DATA)]
+    public partial class WritePlayerIdentityDataSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+
+        public WritePlayerIdentityDataSystem(World world, IECSToCRDTWriter ecsToCRDTWriter) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleEntityDeletionQuery(World);
+            HandleComponentRemovalQuery(World);
+            CreatePlayerIdentityDataQuery(World);
+        }
+
+        // 'PlayerIdentityDataComponent' is put on this scene world entities from the GLOBAL WORLD PlayerComponentsHandlerSystem
+        [Query]
+        [None(typeof(PBPlayerIdentityData), typeof(DeleteEntityIntention))]
+        private void CreatePlayerIdentityData(in Entity entity, PlayerIdentityDataComponent playerIdentityDataComponent)
+        {
+            ecsToCRDTWriter.PutMessage<PBPlayerIdentityData, PlayerIdentityDataComponent>(static (pbPlayerIdentityData, playerIdentityDataComponent) =>
+            {
+                pbPlayerIdentityData.Address = playerIdentityDataComponent.Address;
+                pbPlayerIdentityData.IsGuest = playerIdentityDataComponent.IsGuest;
+            }, playerIdentityDataComponent.CRDTEntity, playerIdentityDataComponent);
+
+            World.Add(entity, new PBPlayerIdentityData
```

Comment:
Instead of `new PBPlayerIdentityData` it should be taken from the pool as you already register this component in the `ComponentsContainer` you don't need to do anything extra related to releasing

Reply:
awsome, implemented here: https://github.com/decentraland/unity-explorer/pull/632/commits/5243d57888c855d02aca207a82446fd3ec4e3123